src/libraries/Common/src/System/Buffers/ArrayBufferWriter.cs

```diff
@@ -186,5 +196,10 @@ private static void ThrowInvalidOperationException_AdvancedTooFar(int capacity)
         {
             throw new InvalidOperationException(SR.Format(SR.BufferWriterAdvancedTooFar, capacity));
         }
+
+        private static void ThrowOutOfMemoryException(uint capacity)
```

Comment:
Because `ArrayBufferWriter` is referenced both in System.Memory and in System.Text.Json, we need to add `BufferMaximumSizeExceeded` to the resx file in System.Text.Json as well.

For example, see `SR.BufferWriterAdvancedTooFar`:
https://github.com/dotnet/runtime/search?q=BufferWriterAdvancedTooFar&unscoped_q=BufferWriterAdvancedTooFar

https://github.com/dotnet/runtime/blob/4a804959512a76556add344d998f3fc9ca9513e8/src/libraries/System.Text.Json/src/Resources/Strings.resx#L309-L311

See the `ArrayBufferWriter.cs` file being included in that project:
https://github.com/dotnet/runtime/blob/4a804959512a76556add344d998f3fc9ca9513e8/src/libraries/System.Text.Json/src/System.Text.Json.csproj#L230-L232

It's used in the Utf8JsonWriter and in S.T.J.Tests (this internal source copy is needed for netstandard2.0):
https://github.com/dotnet/runtime/blob/81bf79fd9aa75305e55abe2f7e9ef3f60624a3a1/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs#L46